hw/top_earlgrey/data/top_earlgrey.h.tpl

```diff
@@ -132,6 +132,41 @@ typedef enum ${plic_targets.name.as_snake_case()} {
 % endfor
 } ${plic_targets.name.as_c_type()};
 
+/**
+ * Alert Handler source peripheral enumeration.
+ *
+ * Enumeration used to determine which peripheral asserted the corresponding
+ * alert.
+ */
+<% alert_sources = helper.alert_sources() %>\
+typedef enum ${alert_sources.name.as_snake_case()} {
+% for (name, value, docstring) in alert_sources.constants:
+  ${name.as_c_enum()} = ${value}, /**< ${docstring} */
+% endfor
+} ${alert_sources.name.as_c_type()};
+
+/**
+ * Alert Handler Alert Ids Enumeration
+ *
+ * Enumeration of all Alert Handler alert IDs. The alert IDs belonging to
```

Comment:
Is the capitalisation intentional here? IDs versus Ids. (You missed the ex Tory leader!)